ibc-testkit/tests/core/ics02_client/create_client.rs

```diff
@@ -86,3 +88,36 @@ fn test_tm_create_client_ok() {
     assert_eq!(expected_client_state.client_type(), client_type);
     assert_eq!(ctx.client_state(&client_id).unwrap(), expected_client_state);
 }
+
+#[test]
+fn test_invalid_frozen_tm_client_creation() {
+    let signer = dummy_account_id();
+
+    let ctx = MockContext::default();
+
+    let router = MockRouter::new_with_transfer();
+
+    let tm_header = dummy_tendermint_header();
+
+    let tm_client_state = dummy_tm_client_state_from_header(tm_header.clone())
+        .inner()
+        .clone()
+        .with_frozen_height(Height::min(0));
+
+    let msg = MsgCreateClient::new(
+        tm_client_state.into(),
+        TmConsensusState::from(tm_header).into(),
+        signer,
+    );
+
+    let msg_envelope = MsgEnvelope::from(ClientMsg::from(msg.clone()));
+
+    let res = validate(&ctx, &router, msg_envelope.clone());
+
+    assert!(res.is_err());
+
+    match res.unwrap_err() {
+        ContextError::ClientError(ClientError::ClientFrozen { .. }) => {}
+        e => panic!("unexpected error: {}", e),
+    }
```

Comment:
you can combine them as

```rs
    match res {
        Err(ContextError::ClientError(ClientError::ClientFrozen { .. })) => {}
        _ => panic!("unexpected response: {}", res),
    }
```

or a one-liner

 `assert!(matches!(res, Err(ContextError::ClientError(ClientError::ClientFrozen { .. }))))`

Reply:
[08aef51](https://github.com/cosmos/ibc-rs/pull/1063/commits/08aef510dd922a7cc474baeaf564a20605a6901f)